core/src/test/java/org/apache/iceberg/util/TestSortOrderUtil.java

```diff
@@ -303,4 +305,33 @@ public void testSortOrderClusteringWithRedundantPartitionFieldsMissing() {
         expected,
         SortOrderUtil.buildSortOrder(table.schema(), updatedSpec, order));
   }
+
+  @Test
+  public void testDropColumnFromDefaultSortOrder() {
+    TestTables.TestTable table =
+        TestTables.create(
+            tableDir, "test", SCHEMA, PartitionSpec.unpartitioned(), SortOrder.unsorted(), 1);
+    table.replaceSortOrder().asc("ts").commit();
+
+    Assertions.assertThatThrownBy(() -> table.updateSchema().deleteColumn("ts").commit())
+        .isInstanceOf(ValidationException.class)
+        .hasMessage("Cannot find source column for sort field: identity(3) ASC NULLS FIRST");
```

Comment:
I think these tests already exist: https://github.com/apache/iceberg/blob/master/core/src/test/java/org/apache/iceberg/TestSortOrder.java#L306-L338

There are `testColumnDropWithSortOrder` and `testIncompatibleSchemaEvolutionWithSortOrder`.

Reply:
Ah, missed those